gobblin-runtime/src/main/java/org/apache/gobblin/runtime/dag_action_store/MysqlDagActionStore.java

```diff
@@ -0,0 +1,172 @@
+  private static final String CREATE_TABLE_STATEMENT = "CREATE TABLE IF NOT EXISTS %s (" +
+  "flow_group varchar(128) NOT NULL, flow_name varchar(128) NOT NULL, flow_execution_id varchar(100) NOT NULL," +
+  "dag_action varchar(100) NOT NULL, modified_time TIMESTAMP DEFAULT CURRENT_TIMESTAMP  on update CURRENT_TIMESTAMP NOT NULL, "
+      + "PRIMARY KEY (flow_group,flow_name,flow_execution_id))";
+
+  @Inject
+  public MysqlDagActionStore(Config config) throws IOException {
+    if (config.hasPath(CONFIG_PREFIX)) {
+      config = config.getConfig(CONFIG_PREFIX).withFallback(config);
+    } else {
+      throw new IOException("Please specify the config for MysqlDagActionStore");
+    }
+    this.tableName = ConfigUtils.getString(config, ConfigurationKeys.STATE_STORE_DB_TABLE_KEY,
+        ConfigurationKeys.DEFAULT_STATE_STORE_DB_TABLE);
+
+    this.dataSource = MysqlStateStore.newDataSource(config);
+    try (Connection connection = dataSource.getConnection();
+        PreparedStatement createStatement = connection.prepareStatement(String.format(CREATE_TABLE_STATEMENT, tableName))) {
+      createStatement.executeUpdate();
+      connection.commit();
+    } catch (SQLException e) {
+      throw new IOException("Failure creation table " + tableName, e);
+    }
+  }
+
+  @Override
+  public boolean exists(String flowGroup, String flowName, String flowExecutionId) throws IOException, SQLException {
+    ResultSet rs = null;
+    try (Connection connection = this.dataSource.getConnection();
+        PreparedStatement existStatement = connection.prepareStatement(String.format(EXISTS_STATEMENT, tableName))) {
+      int i = 0;
+      existStatement.setString(++i, flowGroup);
+      existStatement.setString(++i, flowName);
+      existStatement.setString(++i, flowExecutionId);
+      rs = existStatement.executeQuery();
+      rs.next();
+      return rs.getBoolean(1);
+    } catch (SQLException e) {
+      throw new IOException(String.format("Failure checking existence for table %s of flow with flow group:%s, flow name:%s and flow execution id:%s",
+          tableName, flowGroup, flowName, flowExecutionId), e);
+    } finally {
+      if (rs != null) {
+        rs.close();
+      }
+    }
+  }
+
+  @Override
+  public void addDagAction(String flowGroup, String flowName, String flowExecutionId, DagActionValue dagActionValue)
+      throws IOException {
+    try (Connection connection = this.dataSource.getConnection();
+        PreparedStatement insertStatement = connection.prepareStatement(String.format(INSERT_STATEMENT, tableName))) {
+      int i = 0;
+      insertStatement.setString(++i, flowGroup);
+      insertStatement.setString(++i, flowName);
+      insertStatement.setString(++i, flowExecutionId);
+      insertStatement.setString(++i, dagActionValue.toString());
+      insertStatement.executeUpdate();
+      connection.commit();
```

Comment:
why don't we also return a boolean for this function if the executeUpdate is successful? this should be consistent with the deleteDagAction right

Reply:
As long as add action does not throw exception, add should succeed. But delete can be true or false as even we try to delete some entry that does not exist, it does not throw exception, in this case, we need the boolean result to determine whether we delete the entry or not.